### PR TITLE
feat(nakama): Add ability to register persona tags and sign transactions

### PR DESCRIPTION
**Pull request by jerargus**
Closes: #NAK-3, NAK-4

## What is the purpose of the change

Update the sample Nakama code to:
- Associate a PersonaTag with a Nakama UserID
- Generate and store a PrivateKey on startup (or load an existing private key)
- Sign transactions with the generated Private key and PersonaTag associated with the UserID.

## Testing and Verifying

The code can be manually tested using the instructions in the "Claim PersonaTags and sign transactions" section of the README.

TODO: Implement automatic integration tests. See https://linear.app/arguslabs/issue/NAK-4/write-integration-tests-for-nakama-account-management

## Documentation and Release Note

- Does this pull request introduce a new feature or user-facing behavior changes? (yes)
- Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
- How is the feature or change documented? (The README.md)

### CODE REVIEW

**Comment by technicallyty** [comment]
will get back to this on monday.

File: game/sample_game/nakama/private_key.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/heroiclabs/nakama-common/runtime"
+)
+
+var (
+	ErrorNoStorageObjectFound       = errors.New("no storage object found")
+	ErrorTooManyStorageObjectsFound = errors.New("too many storage objects found")
+
+	globalPrivateKey    *ecdsa.PrivateKey
```
Comment (technicallyty) [inline game/sample_game/nakama/private_key.go:24]: i think there are security implications that come with storing pk in memory, we may want to open a ticket to address.

File: game/sample_game/nakama/private_key.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	}
+	var pkObj privateKeyStorageObj
+	if err := json.Unmarshal([]byte(objs[0].Value), &pkObj); err != nil {
+		return "", err
+	}
+	return pkObj.Value, nil
+}
+
+// setOnePKStorageObj saves the given value to the privateKeyCollection in Nakama's storage layer.
+func setOnePKStorageObj(ctx context.Context, nk runtime.NakamaModule, key, value string) error {
```
Comment (technicallyty) [inline game/sample_game/nakama/private_key.go:65]: can we rename `key` and `value` to reflect what we actually need to pass into the function?

File: game/sample_game/nakama/private_key.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	adminAccountID       = "00000000-0000-0000-0000-000000000000"
+)
+
+type privateKeyStorageObj struct {
+	Value string
+}
+
+// getOnePKStorageObj loads one specific runtime.StorageObject from the privateKeyCollection in
+// Nakama's storage layer. An error is returned if too few or too many storage objects are found.
+func getOnePKStorageObj(ctx context.Context, nk runtime.NakamaModule, key string) (string, error) {
```
Comment (technicallyty) [inline game/sample_game/nakama/private_key.go:43]: can we rename `key` to reflect what we actually need to pass into the function?

File: game/sample_game/nakama/cardinal.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+	readPersonaSignerStatusUnknown   = "unknown"
+	readPersonaSignerStatusAvailable = "available"
+	readPersonaSignerStatusAssigned  = "assigned"
+
+	ErrorPersonaSignerNotAssigned = errors.New("persona signer has not been assigned")
+)
+
+func makeURL(resource string) string {
+	addr := os.Getenv(EnvCardinalAddr)
```
[the remark this answers is not in the document]
Comment (jerargus) [reply game/sample_game/nakama/cardinal.go]: I moved it into Nakama's InitModule pattern

File: game/sample_game/nakama/cardinal.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+		return 0, fmt.Errorf("create persona failed with status %q", s)
+	}
+	return createPersonaResponse.Tick, nil
+}
+
+func cardinalQueryPersonaSigner(ctx context.Context, personaTag string, tick int) (signerAddress string, err error) {
+	buf, err := json.Marshal(map[string]interface{}{
+		"PersonaTag": personaTag,
+		"Tick":       tick,
+	})
```
[the remark this answers is not in the document]
Comment (jerargus) [reply game/sample_game/nakama/cardinal.go]: Done

File: game/sample_game/nakama/private_key.go (hunk 1 above)
Comment (jerargus) [reply game/sample_game/nakama/private_key.go:27]: Added ticket: https://linear.app/arguslabs/issue/NAK-5/review-the-pattern-of-storing-the-nakama-private-key-in-a-global

**Review by technicallyty — APPROVED** [review]
lgtm

**Pull request merged**